hazelcast/src/main/java/com/hazelcast/map/impl/AbstractMapServiceContextSupport.java

```diff
@@ -152,8 +152,14 @@ public void interceptAfterRemove(String mapName, Object value) {
     }
 
     @Override
-    public String addInterceptor(String mapName, MapInterceptor interceptor) {
-        return mapServiceContext.getMapContainer(mapName).addInterceptor(interceptor);
+    public void addInterceptor(String id, String mapName, MapInterceptor interceptor) {
+        mapServiceContext.getMapContainer(mapName).addInterceptor(id, interceptor);
```

Comment:
This is called a train wreck. Every word is a wagon and every dot is a connection between the train wagons. One of the problems with train wrecks is that if you get a NPE in production, you have no idea which wagon caused it.

I see that this style is often being used, especially in the map, but it is a well known bad practice.

http://blog.shamess.info/2012/05/31/clean-code-and-train-wrecks/